pypeit/core/findobj_skymask.py

```diff
@@ -1244,146 +1244,153 @@ def objs_in_slit(image, ivar, thismask, slit_left, slit_righ,
-    ivar_smash = utils.inverse(var_smash)*gpm_smash
-    snr_smash = flux_smash_recen*np.sqrt(ivar_smash)
-
-    # Smooth this SNR image with a Gaussian set by the input fwhm
-    gauss_smth_sigma = (fwhm/2.3548)
-    snr_smash_smth = scipy.ndimage.filters.gaussian_filter1d(snr_smash, gauss_smth_sigma, mode='nearest')
-    flux_smash_smth = scipy.ndimage.filters.gaussian_filter1d(flux_smash_recen, gauss_smth_sigma, mode='nearest')
-    # Search for spatial direction peaks in the smoothed snr image
-    _, _, x_peaks_out, x_width, x_err, igood, _, _ = arc.detect_lines(
-        snr_smash_smth, input_thresh=snr_thresh, fit_frac_fwhm=1.5, fwhm=fwhm, min_pkdist_frac_fwhm=0.75,
-        max_frac_fwhm=10.0, cont_subtract=False, debug_peak_find=False)
-
-    x_peaks_all = x_peaks_out[igood]
-    #x_peaks_all = arc.detect_peaks(snr_smash_smth, mph=snr_thresh, mpd=fwhm*0.75, show=False)
-    snr_peaks_all = np.interp(x_peaks_all, np.arange(nsamp), snr_smash_smth)
-    flux_peaks_all = np.interp(x_peaks_all, np.arange(nsamp), flux_smash_smth)
-    npeaks_all = len(x_peaks_all)
-
-    near_edge_bpm = (x_peaks_all < trim_edg[0]) | (x_peaks_all > (nsamp - trim_edg[1]))
-    npeak_not_near_edge = np.sum(np.logical_not(near_edge_bpm))
-
-    if np.any(near_edge_bpm):
-        msgs.warn('Discarding {:d}'.format(np.sum(near_edge_bpm)) +
-                  ' at spatial pixels spat = {:}'.format(x_peaks_all[near_edge_bpm]) +
-                  ' which land within trim_edg = (left, right) = {:}'.format(trim_edg) +
-                  ' pixels from the slit boundary for this nsamp = {:5.2f}'.format(nsamp) + ' wide slit')
-        msgs.warn('You must decrease from the current value of trim_edg in order to keep them')
-        msgs.warn('Such edge objects are often spurious')
-
-
-    # If the user requested the nperslit most significant peaks have been requested, then only return these
-    if nperslit is not None:
-        # If the requested number is less than (the non-edge) number found, mask them out
-        if nperslit < npeak_not_near_edge:
-            snr_peaks_not_edge = np.sort(snr_peaks_all[np.logical_not(near_edge_bpm)])[::-1]
-            snr_thresh_perslit = snr_peaks_not_edge[nperslit-1]
-            nperslit_bpm = np.logical_not(near_edge_bpm) & (snr_peaks_all < snr_thresh_perslit)
+    # Return if none found and no hand extraction
+    if not np.any(gpm_smash): 
+        sobjs = specobjs.SpecObjs()
+        if hand_extract_dict is None:
+            # Instantiate a null specobj and return
+            msgs.info('No objects found automatically.  Consider manual extraction.')
+            return sobjs
         else:
-            nperslit_bpm = np.zeros(npeaks_all, dtype=bool)
+            msgs.info('No objects found automatically.')
+            
     else:
-        nperslit_bpm = np.zeros(npeaks_all, dtype=bool)
+        # Compute the formal corresponding variance over the set of pixels that are not masked by gpm_sigclip
```

Comment:
We don't necessarily have to do so for this PR, but this function really needs to be broken up.  There are now more than 100 lines of code in this `else` block.

Reply:
Agreed, and I considered it a bit.

But the entire extraction codes need refactoring.  
And that's a bigger job than this PR..